content/en/docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -95,7 +140,31 @@ Ensure that the KMS plugin runs on the same host(s) as the Kubernetes master(s).
 
 To encrypt the data:
 
-1. Create a new encryption configuration file using the appropriate properties for the `kms` provider:
+1. Create a new encryption configuration file using the appropriate properties for the `kms` provider to encrypt resources like Secrets and ConfigMaps:
+
+  **KMS v1**
+
+   ```yaml
+   apiVersion: apiserver.config.k8s.io/v1
+   kind: EncryptionConfiguration
+   resources:
+     - resources:
+         - secrets
+       providers:
+         - kms:
+             name: myKmsPluginFoo
+             endpoint: unix:///tmp/socketfile.sock
+             cachesize: 100
+             timeout: 3s
+         - identity: {}
+         - kms:
+             name: myKmsPluginBar
+             endpoint: unix:///tmp/socketfile.sock
+             cachesize: 100
+             timeout: 3s
+         - identity: {}
```

Comment:
Technically one should leave this off if they know all data is stored as encrypted because then having write access to etcd without access to the key vault means that one cannot inject data into the store (i.e. you need either control over the API server config or control over etcd+key vault to do harm).

Reply:
I believe only one identity entry might be present/required and as a last.
Due to the sequential nature of the api server to try each providers, we would first assume data field payload has been encrypted with the main kms, then maybe a former kms from which we want to move away from by doing a replace operation would still be present, and finally the identity provider for pre-kms plugin secrets that we might not have replaced.

Reply:
I understand how the code works, I am pointing out that by recommending the use of the identity provider, we limit the already small value of the encryption at rest support in Kube but allowing an attacker to inject data simply if they have access to etcd.  We should stop doing that because part of the threat model for KMS is not trusting etcd.